### ISSUE TITLE
Question: Broken Texas Instruments CC26X2R1 info ?

### ISSUE THREAD

**Issue by nikkolade** [Koenkk/Z-Stack-firmware#171]
I know that this is not directly related to Z-Stack-Firmware, but I wonder if somebody else has some information related to this kind of problematic HW case.

A Texas Instruments CC26X2R1 card that has been working just fine since last fall, suddenly broke down and is not anymore being detected as Texas Instruments CC26X2R1, but instead XDS110 (03.00.00.05) Embed with CMSIS-DAP.

Is this card still salvageable without additional HW ? I don't own any additional HW debuggers or such, so can this possibly be re-flashed somewhow ?

I can see from the logs that once this device was briefly disconnected from the host and then
reconnected, there were below error messages:

> May 14 18:57:24 host kernel: [1248816.014792] usb 6-2: USB disconnect, device number 2
May 14 18:57:24 host kernel: [1248816.014859] cdc_acm 6-2:1.0: acm_ctrl_irq - usb_submit_urb failed: -19
May 14 18:57:25 host kernel: [1248816.963304] usb 6-2: new high-speed USB device number 3 using xhci_hcd
May 14 18:57:25 host kernel: [1248817.091110] usb 6-2: config index 0 descriptor too short (expected 219, got 56)
May 14 18:57:25 host kernel: [1248817.091113] usb 6-2: invalid descriptor for config index 0: type = 0x2, length = 0
May 14 18:57:25 host kernel: [1248817.091117] usb 6-2: can't read configurations, error -22
May 14 18:57:25 host kernel: [1248817.205303] usb 6-2: new high-speed USB device number 4 using xhci_hcd
May 14 18:57:25 host kernel: [1248817.320345] usb 6-2: device descriptor read/64, error -71
May 14 18:57:26 host kernel: [1248817.553362] usb 6-2: New USB device found, idVendor=0451, idProduct=bef3, bcdDevice= 1.00
May 14 18:57:26 host kernel: [1248817.553365] usb 6-2: New USB device strings: Mfr=1, Product=2, SerialNumber=3
May 14 18:57:26 host kernel: [1248817.553367] usb 6-2: Product: XDS110 (03.00.00.05) Embed with CMSIS-DAP
May 14 18:57:26 host kernel: [1248817.553368] usb 6-2: Manufacturer: Texas Instruments
May 14 18:57:26 host kernel: [1248817.553369] usb 6-2: SerialNumber: L1100BV7
May 14 18:57:26 host kernel: [1248817.563122] hid-generic 0003:0451:BEF3.0004: hiddev0,hidraw0: USB HID v1.11 Device [Texas Instruments XDS110 (03.00.00.05) Embed with CMSIS-DAP] on usb-0000:00:10.0-2/input5

I saw that there has obviously been at least one similar case in the past, but I wonder if that device was ever fixed or not.

Failed to flash CC26X2R1 with Z-Stack 3.0 #150

**Comment by nikkolade** [comment]
Does anyone have ideas, how to rewrite/flash the empty usb idProduct field ?
The problematic device is being identified like this:

Bus 006 Device 005: ID 0451:bef3 Texas Instruments, Inc. 

while the second, still functioning device shows up like this:

Bus 001 Device 008: ID 0451:bef3 Texas Instruments, Inc. CC1352R1 Launchpad

I already used the CCS v10 and re-flashed the XDS110 firmware without any benefit.
Also the linux Uniflash worked just fine, when flashing the Zigbee2mqtt firmware,
although the uniflash wasn't able to autodetect the device, most likely due to this:
------------------------------------------------------------------------------------------------
Bus 006 Device 005: ID 0451:bef3 Texas Instruments, Inc. 
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               1.10
  bDeviceClass          239 Miscellaneous Device
  bDeviceSubClass         2 ?
  bDeviceProtocol         1 Interface Association
  bMaxPacketSize0        64
  idVendor           0x0451 Texas Instruments, Inc.
  idProduct          0xbef3 
  bcdDevice            1.00
  iManufacturer           1 Texas Instruments
  iProduct                2 XDS110 (03.00.00.12) Embed with CMSIS-DAP
  iSerial                 3 L1100BV7
  bNumConfigurations      1
------------------------------------------------------------------------------------------------

While the working device shows this:
------------------------------------------------------------------------------------------------

Bus 001 Device 008: ID 0451:bef3 Texas Instruments, Inc. CC1352R1 Launchpad
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               1.10
  bDeviceClass          239 Miscellaneous Device
  bDeviceSubClass         2 
  bDeviceProtocol         1 Interface Association
  bMaxPacketSize0        64
  idVendor           0x0451 Texas Instruments, Inc.
  idProduct          0xbef3 CC1352R1 Launchpad
  bcdDevice            1.00
  iManufacturer           1 
  iProduct                2 
  iSerial                 3 
  bNumConfigurations      1
------------------------------------------------------------------------------------------------

**Comment by nikkolade** [comment]
Finally managed to recover the device into usable state.

Originally, the device got some sort of SW corruption due to removal of the usb-wire
at the wrong moment (apparently) and the device stopped showing up as
CC26X2R1 Launchpad. No matter what I tried, the uniflash or the CCS 10
xdsdfu flashing wouldn't bring it back to usable state.

While trying to get the device back into working state, it first went all downhill
and got into a situation, where the device was showing like this:

1cbe:02a5 Luminary Micro Inc. XDS110 (03.00.00.12) Embed with CMSIS-DAP

And after this it wasn't possible to even try accessing the device via xdsdfu.
In the end I had to follow the instructions of the "XDS110SupportReadMe.pdf"
(from the CCS 10 package) that told to reset the chip by ground the pin 97
of the main chip.

After the device was back in flash mode, I was able to flash the XDS110 firmware
from the CCS 9.3. If I tried to do this step using the  CCS 10, the device went back
to coma.

The succesful path was:
1. Reset the device physically by grounding the pin 97.
2. Use CCS 9.3 xdsdfu to flash the XDS110 firmware.
   "xdsdfu -r firmware.bin -r"
    ---> The device recognized properly as CC26X2R1 launchpad.
3. Fetch the latest uniflash 6 and use that one
   --> Update the XDS110 firmware succesfully and flash the
         zigbee2mqtt firmware.

Trying to use CCS 10 instead of the CCS 9 only got the device back to
square one. Using the uniflash 5.2 to upgrade the XDS110 firmware before
zigbee2mqtt flash, failed and got the device back to square one.

I was never able to use the "xdsdfu -b boot_loader.bin" as the command
displayed (Y/N) prompt, but didn't allow answering to it.
